solr/core/src/java/org/apache/solr/handler/IndexFetcher.java

```diff
@@ -677,12 +677,12 @@ IndexFetchResult fetchLatestIndex(boolean forceReplication, boolean forceCoreRel
         markReplicationStop();
         return successfulInstall ? IndexFetchResult.INDEX_FETCH_SUCCESS : IndexFetchResult.INDEX_FETCH_FAILURE;
       } catch (ReplicationHandlerException e) {
-        log.error("User aborted Replication");
+        log.error("User aborted Replication", e);
         return new IndexFetchResult(IndexFetchResult.FAILED_BY_EXCEPTION_MESSAGE, false, e);
       } catch (SolrException e) {
         throw e;
       } catch (InterruptedException e) {
-        throw new InterruptedException("Index fetch interrupted");
+        throw (InterruptedException)(new InterruptedException("Index fetch interrupted").initCause(e));
```

Comment:
observation: changing the details of the exception being thrown here (not its type but it having or not having a cause or a particular type of cause) could theoretically change the calling code's behaviour. practically that seems unlikely though. also inspection of calling code finds no obvious signs that behaviour would change.